Shard python_torch_functions.cpp

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62188
* __->__ #62187
* #62186
* #62278
* #62185
* #62184

This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

Differential Revision: [D29962048](https://our.internmc.facebook.com/intern/diff/D29962048)

peterbell10 committed:
Shard python_torch_functions.cpp

This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

albanD commented on torch/csrc/autograd/python_torch_functions_manual.cpp:
```diff
@@ -0,0 +1,726 @@
+
+namespace torch { namespace autograd {
+
+inline Tensor dispatch_arange(const Scalar& end, Tensor result) {
```
Do you know why we have so many of these bindings that are manual?

peterbell10 replied:
Not sure. I see that `arange` specifically was manually bound in #7016.

albanD replied:
cc @gchanan do you remember if this is still valid? :)

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29962048).

albanD reviewed (approved):
SGTM

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

Differential Revision: [D29962048](https://our.internmc.facebook.com/intern/diff/D29962048)

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

Differential Revision: [D29962048](https://our.internmc.facebook.com/intern/diff/D29962048)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29962048).

peterbell10 committed:
Update on "Shard python_torch_functions.cpp"


This file can take 3 minutes on its own to compile, and after
python_functions.cpp is the second limiting factor for compile time of
`libtorch_python` on a 32-core threadripper. This splits it into 3 files that
take around 1 minute each to compile.

Differential Revision: [D29962048](https://our.internmc.facebook.com/intern/diff/D29962048)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29962048).
